appsec/src/helper/remote_config/http_api.cpp

```diff
@@ -75,16 +90,41 @@ std::string execute_request(const std::string &host, const std::string &port,
             "Connection error - " + err + " - " + e.what());
     }
 
-    return result;
+    co_return result;
 }
 
+std::string execute_request_sync(const std::string &host,
+    const std::string &port, const http::request<http::string_body> &req)
+{
+
+    net::io_context ioc;
+    net::awaitable<std::string> client_coroutine =
+        execute_request(host, port, req);
+
+    std::promise<std::string> promise;
+    auto fut = promise.get_future();
+
+    net::co_spawn(ioc, std::move(client_coroutine),
+        [&](const std::exception_ptr &eptr, std::string body) {
+            if (eptr) {
+                promise.set_exception(eptr);
+            } else {
+                promise.set_value(std::move(body));
+            }
+        });
+
+    ioc.run();
```

Comment:
This is all very interesting... but what did we gain aside from having a timeout?

Reply:
The whole point is having a timeout. Having no timeout in a network operation, in a daemon on top of it, is unacceptable, as the operation can block indefinitely, preventing further communication.

Reply:
I understand the purpose of the timeout of course, I was more interested in any other potential benefits if any.